pkg/k8shandler/configmaps.go

```diff
@@ -46,6 +46,52 @@ type indexSettingsStruct struct {
 	ReplicaShards string
 }
 
+// CreateOrUpdateConfigMap reconciles a configmap
+func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateConfigMap(cm *v1.ConfigMap) error {
+	err := elasticsearchRequest.client.Create(context.TODO(), cm)
+	if err == nil {
+		logrus.Debug("created configmap %q in namespace $q", cm.Name, cm.Namespace)
+		return nil
+	}
+	if !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failure constructing ConfigMap: %w", err)
+	}
+
+	// Get existing configMap to check if it is same as what we want
+	current := cm.DeepCopy()
+	err = elasticsearchRequest.client.Get(context.TODO(), types.NamespacedName{Name: current.Name, Namespace: current.Namespace}, current)
```

Comment:
fyi we already have a function in this file: `getConfigmap`

Reply:
getConfigmap also swallows an error and that seems _very_ dangerous to me.